iceoryx2-ffi/cxx/include/iox/slice.hpp

```diff
@@ -25,29 +26,126 @@ class Slice {
     using ConstIterator = const T*;
     using ValueType = T;
 
-    auto size() const -> uint64_t {
-        IOX_TODO();
-    }
-    auto operator[](const uint64_t n) const -> const T& {
-        IOX_TODO();
-    }
-    auto operator[](const uint64_t n) -> T& {
-        IOX_TODO();
-    }
-    auto begin() -> Iterator {
-        IOX_TODO();
-    }
-    auto begin() const -> ConstIterator {
-        IOX_TODO();
-    }
-    auto end() -> Iterator {
-        IOX_TODO();
-    }
-    auto end() const -> ConstIterator {
-        IOX_TODO();
-    }
+    Slice(T* data, uint64_t number_of_elements);
+    Slice(const T* data, uint64_t number_of_elements);
```

Comment:
@elfenpiff Here is my attempt to address this: https://github.com/eclipse-iceoryx/iceoryx2/pull/492/commits/39858c83dbd9f9fe408ab37edb03d68e4a0c82cd

1. I kept the `Slice<T>` signature
2. If `T` is `const`, it should return `const T*` or `const T&` depending on the method, and vice versa
3. I made. some aliases `MutableSlice` and `ImmutableSlice` for readability purposes, intention is to make it less error prone (i.e. forgetting to make T const when the semantics require it)
4. I kept the name `Slice` for the implementation so that at the service level, the user only needs to specify the service type of `Slice<MyType>`